AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1635,6 +1644,20 @@ - (void)recursivelyClearFetchedData
   [self clearFetchedData];
 }
 
+/**
+ * We currently only set interface state on nodes
+ * in table/collection views. For other nodes, if they are
+ * in the hierarchy we return `Unknown`, otherwise we return `None`.
+ *
+ * TODO: Avoid traversing up node hierarchy due to possible deadlock.
+ * @see https://github.com/facebook/AsyncDisplayKit/issues/900
+ * Possible solution is to push `isInCellNode` state downward on `addSubnode`/`removeFromSupernode`.
+ */
+- (BOOL)supportsInterfaceState {
```

Comment:
This is good as a prototype implementation, but we indeed need to improve efficiency with something like ASHierarchyState to avoid the repeated recursion or measurable performance impact will occur.  I think we'll land as-is once I am able to start on that change, but it would be important to fix before releasing (or even having in master very long, definitely increases deadlock probability)